javascript/javascript1/week2/smartease.js

```diff
@@ -0,0 +1,177 @@
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+
+  return weekDays[(today + daysAfter) % 7];
+}
+const daysAfter = 4;
+const eventDay = getEventWeekday(daysAfter);
+console.log(eventDay);
+
+//Weather wear
+console.log("**********************");
+console.log("Ex 4: Weather wear");
+console.log("**********************");
+
+function selectClothes(temperature) {
+  if (temperature <= 5) {
+    return "Winter Jacket and Thermal Wear";
+  } else if (temperature > 5 && temperature < 18) {
+    return "Normal Jacket and Pant and T-shirt";
+  } else if (temperature >= 18) {
+    return "Shorts and a T-shirt";
+  } else {
+    return "Please provide the current temperature in degrees";
+  }
+}
+const temperature = 18;
+const clothesToWear = selectClothes(temperature);
+console.log(clothesToWear);
+
+//Student manager
+console.log("**********************");
+console.log("Ex 5: Student manager");
+console.log("**********************");
+let class07Students = [];
+function addStudentToClass(studentName) {
+  studentName = studentName.toLowerCase();
+  if (
+    (class07Students.length < 6 &&
+      !class07Students.includes(studentName) &&
+      studentName !== "") ||
+    studentName === "mary elizabeth"
+  ) {
+    class07Students.push(studentName);
+  } else if (studentName === "") {
+    console.log("Please provide the student name");
+  } else if (class07Students.includes(studentName)) {
+    console.log(`Student ${studentName} is already in the class`);
+  } else {
+    console.log("Cannot add more students to class 07");
+  }
+  return class07Students;
+}
+addStudentToClass("Benjamin");
+addStudentToClass("Hughes");
+addStudentToClass("Josefine");
+addStudentToClass("Henrik");
+addStudentToClass("Ida");
+addStudentToClass("Frede");
+addStudentToClass("Mary Elizabeth");
+addStudentToClass("Hello");
+addStudentToClass("benjamin");
+addStudentToClass("");
+
+console.log(class07Students);
+
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+console.log(getNumberOfStudents());
+
+//Candy helper optional
+console.log("**********************");
+console.log("Ex 6: Candy helper optional");
+console.log("**********************");
+let price = 0;
+function addCandy(candyType, weight) {
+  candyType = candyType.toLowerCase();
+  if (candyType === "sweet") {
+    price = weight * 0.5;
+  } else if (candyType === "chocolate") {
+    price = weight * 0.7;
+  } else if (candyType === "toffee") {
+    price = weight * 1.1;
+  } else if (candyType === "chewing-gum") {
+    price = weight * 0.03;
+  } else {
+    console.log(
+      "Please select the candyType as Sweet or Chocolate or Toffee or Chewing-gum"
+    );
+  }
+  return (price = Math.round(price));
+}
+let boughtCandyPrices = [];
+addCandy("sweet", 15);
+boughtCandyPrices.push(price);
+addCandy("toffee", 20);
+boughtCandyPrices.push(price);
+addCandy("chocolate", 15);
+boughtCandyPrices.push(price);
+console.log(boughtCandyPrices);
```

Comment:
You are mixing up return values and outer scoped variables.

Try to read the task again. I would do something like this:
```js
const boughtCandyPrices = [];
function addCandy(candyType, weight) {
  candyType = candyType.toLowerCase();
  let price = 0;
  if (candyType === "sweet") {
    price = weight * 0.5;
  } else if (candyType === "chocolate") {
    price = weight * 0.7;
  } else if (candyType === "toffee") {
    price = weight * 1.1;
  } else if (candyType === "chewing-gum") {
    price = weight * 0.03;
  } else {
    console.log(
      "Please select the candyType as Sweet or Chocolate or Toffee or Chewing-gum"
    );
  }
  boughtCandyPrices.push(price);
}
```
And then add all the candys.

Reply:
Yes, I misunderstood the concept of the task initially. However, I've since revised the code according to your feedback